helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -204,4 +236,28 @@ private MappingCalculator getMappingCalculator(Rebalancer rebalancer, String res
 
     return mappingCalculator;
   }
+
+  class JobResourcePriority implements Comparable<JobResourcePriority> {
```

Comment:
From my point of view, this is a priority with a job reference. So call it "priority" is fine.
JobResourceWithPriority is too verbose.
And ComparableJobResource/JobResource implies that it is another kind of job resource class. They seem to be even more confusing.